common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -209,9 +246,16 @@ private AppShufflePartitionInfo getOrCreateAppShufflePartitionInfo(
         appShuffleInfo.getMergedShuffleIndexFilePath(shuffleId, shuffleMergeId, reduceId));
       File metaFile =
         appShuffleInfo.getMergedShuffleMetaFile(shuffleId, shuffleMergeId, reduceId);
+      // Make sure unuseful non-finalized merged data/index/meta files get cleaned up
+      // during service restart
+      if (dataFile.exists()) dataFile.delete();
+      if (indexFile.exists()) indexFile.delete();
+      if (metaFile.exists()) metaFile.delete();
       try {
-        return newAppShufflePartitionInfo(appShuffleInfo.appId, shuffleId, shuffleMergeId,
-          reduceId, dataFile, indexFile, metaFile);
+        AppShufflePartitionInfo appShufflePartitionInfo =
+            newAppShufflePartitionInfo(appShuffleInfo.appId, appShuffleInfo.attemptId,
```

Comment:
How about changing `newAppShufflePartitionInfo` to accept just `appShuffleInfo` just so that we can avoid more changes to this method signature in future.